scripts/ai/AIDriveStrategyFindBales.lua

```diff
@@ -483,13 +488,15 @@ function AIDriveStrategyFindBales:approachBale()
         if not self:isReadyToLoadNextBale() then
             self:debug('Start picking up bale')
             self.state = self.states.WORKING_ON_BALE
+            self:findBales() --- Refreshes for bale counter

```

Comment:
Just a bit worried about the performance here with big fields and many bales, one could just decrement self.balesLeft until we'd look for new bales anyway, but may not be a problem at all.